packages/ai-jsx/src/batteries/docs.tsx

```diff
@@ -763,6 +766,20 @@ export interface DocsQAProps<ChunkMetadata extends Jsonifiable = Jsonifiable> {
    * ```
    */
   chunkFormatter?: (props: { doc: ScoredChunk<ChunkMetadata> }) => Node;
+
+  /**
+   * The component used to format results from a DocsQAWithSources query.
+   *
+   * ```tsx
+   *   function FormatQAResult(result: QAWithSourcesResult) {
+   *     if (result.sources?.length) {
+   *       return `${result.answer}\n\nSources:\n${result.sources.join('\n')}`;
+   *     }
+   *     return result.answer;
+   *   }
+   * ```
+   */
+  resultFormatter?: (result: QAWithSourcesResult) => Node;
```

Comment:
`DocsQA` doesn't use `resultFormatter`, right? But it does take `DocsQAProps` as props? We shouldn't have an unused field in the props for `DocsQA`.

Instead, I'd do something like:

```tsx
interface DocsQAWithSourcesProps extends DocsQAProps {
  resultFormatter?: (result: QAWithSourcesResult) => Node;
}
```